named_value_checker.go

```diff
@@ -0,0 +1,26 @@
+// +build go1.9
```

Comment:
With files separated by Go version compatibility it's rather hard to see at the first glance that there are things dependent on it.

Looking at the project tree:
```
conn.go
connector.go
driver.go
name_value_checker.go
sql.go
...
```

It's rather hard to get the gist of it, especially if one is not familiar with the fact that `NamedValueChecker` is in Go since 1.9.

Therefore, it's usual in Go libraries to have version specific code placed in a corresponding file with version suffix.
If you would be to extend driver to implement `DriverContext` and the driver-related struct and other bits are in `driver.go` we would get `driver_go110.go`.
Extending this to everything else:
```
conn.go
conn_go19.go // NameValueChecker impl goes here
conn_go110.go // SessionResetter impl goes here
driver.go
driver_go110.go // DriverContext impl goes here
stmt.go
stmt_go19.go // NamedValueChecker as well
```